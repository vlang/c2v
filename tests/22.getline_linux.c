#include <stdio.h>
#include <stdlib.h>

int main() {
    char *input = NULL;
    size_t len = 0;
    ssize_t read;
    printf("Enter text (Ctrl+D to quit):\n");
    read = getline(&input, &len, stdin);

    if (read != -1) {
        printf("Entered: %s", input);
    } else {
        printf("error reading input\n");
    }

    free(input);
    return 0;
}