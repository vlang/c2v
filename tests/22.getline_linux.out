@[translated]
module main

fn C.getline(__lineptr &&u8, __n &usize, __stream &C.FILE) isize

struct Lldiv_t {
	quot i64
	rem  i64
}

struct Random_data {
	fptr      &int
	rptr      &int
	state     &int
	rand_type int
	rand_deg  int
	rand_sep  int
	end_ptr   &int
}

struct Drand48_data {
	__x     [3]u16
	__old_x [3]u16
	__c     u16
	__init  u16
	__a     i64
}

fn main() {
	input := (unsafe { nil })
	len := 0
	read := isize(0)
	C.printf(c'Enter text (Ctrl+D to quit):\n')
	read = C.getline(&input, &len, C.stdin)
	if read != -1 {
		C.printf(c'Entered: %s', input)
	} else {
		C.printf(c'error reading input\n')
	}
	C.free(input)
	return
}
